tool/tsh/common/recording_export.go

```diff
@@ -119,7 +120,19 @@ loop:
 			switch evt := evt.(type) {
 			case *apievents.WindowsDesktopSessionStart:
 			case *apievents.WindowsDesktopSessionEnd:
-				break loop
+				if !fastPathReceived {
+					break loop
+				}
+				if movie != nil {
+					movie.Close()
+					os.Remove(currentFilename)
+				}
+				url := fmt.Sprintf("https://%s/web/cluster/%s/session/%s?recordingType=desktop&durationMs=%d",
+					webProxyAddr,
+					evt.ClusterName,
+					evt.SessionID,
+					lastEmitted)
```

Comment:
Instead of using `lastEmitted`, you could compute the duration by looking at the `StartTime` and `EndTime` fields of the `WindowsDesktopSessionEnd` event.

This would be more correct in cases where there was idle time prior to the session end event.

Reply:
That's better, I've changed it, thanks!